docs/rhel_dpdk_container_disk_image.md

```diff
@@ -0,0 +1,153 @@
+# Building RHEL DPDK VM ContainerDisk Image
+
+Create a new RHEL 8.7 virtual machine, with at least the following specifications:
+
+- 2 CPU Cores
+- 4 GiB RAM
+- 20 GB of free space in `/var`
+
+In order to build the ContainerDisk image, the solution uses a pipeline of:
+
+- [image-builder](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/composing_a_customized_rhel_system_image/index)
+- [virt-customize](https://www.libguestfs.org/virt-customize.1.html)
+- [podman](https://podman.io)
+
+Subscribe the VM using subscription-manager:
+
+```bash
+sudo subscription-manager register
+```
+
+## Install build dependencies
+
+```bash
+sudo dnf install libguestfs-tools
+sudo dnf install podman
+sudo dnf install osbuild-composer composer-cli bash-completion
+sudo systemctl enable --now osbuild-composer.socket
+source /etc/bash_completion.d/composer-cli
+```
+
+Add the user to the `weldr` group in order to use `composer-cli` without `sudo`:
+
+```bash
+newgrp weldr
+sudo usermod -a -G weldr <user>
+```
+
+Verify that the image builder is working:
+
+```bash
+composer-cli status show
+```
+
+Verify that you can use RHEL 8.7 distro:
+
+```bash
+composer-cli distros list
+```
+
+## Build the base image
+
+Create the blueprint file:
+
+```bash
+cat << EOF > dpdk-vm.toml
+name = "dpdk_image"
+description = "Image to use with the DPDK checkup"
+version = "0.0.1"
+distro = "rhel-87"
+
+[[packages]]
+name = "dpdk"
+
+[[packages]]
+name = "dpdk-tools"
+
+[[packages]]
+name = "driverctl"
+
+[[packages]]
+name = "tuned-profiles-cpu-partitioning"
+
+[customizations.kernel]
+append = "default_hugepagesz=1GB hugepagesz=1G hugepages=8 isolcpus=2-7"
+
+[customizations.services]
+disabled = ["NetworkManager-wait-online", "sshd"]
+EOF
+```
+
+Push the blueprint file:
+
+```bash
+composer-cli blueprints push dpdk-vm.toml
+```
+
+Start building the image:
+
+```bash
+composer-cli compose start dpdk_image qcow2
+```
+
+Periodically check for the build status, and wait for it to be `FINISHED`:
+
+```bash
+composer-cli compose list
+```
+
+Get the ready qcow2 image:
+
+```bash
+composer-cli compose image <UID>
+```
+
+## Further image customization
```

Comment:
I think "Further custom.." is a bit redundant.

Reply:
Changed to "Base Image Customization"